### PR TITLE
Fix checking Origin header for WebSockets

### PR DESCRIPTION
**Pull request by victorbalan**
* Fix check origin
 * Remove the wc.close from the reader

### CODE REVIEW

File: ws/ws.go [hunk 1]
```diff
@@ -99,7 +100,6 @@ func Write(key *datastore.Key, buf []byte) error {
 }
 
 func reader(ws *websocket.Conn) {
-	defer ws.Close()
```
Comment (flowlo) [inline ws/ws.go]: Why did you remove this? Could potentially leak `*websocket.Conn`s

**Comment by flowlo** [comment]
I object to changing the container output in any way. Not even removing a trailing newline. If you did this because you were getting length mismatches, we should rather look at out reference output. It should have a trailing newline too.

File: ws/ws.go (hunk 1 above)
Comment (victorbalan) [inline ws/ws.go]: I removed this because after a few runs nothing arrives in the frontend. We could close the ws programatically after some time.

**Comment by victorbalan** [comment]
you`re right, i`ll modify the test

File: ws/ws.go (hunk 1 above)
Comment (flowlo) [inline ws/ws.go]: Have you tried logging whether the connection was closed, to be sure that this is the cause?

````go
defer func() {
	// Log "Closing WebSocket".
	ws.Close()
}
````

**Comment by flowlo** [comment]
Well, only modify it, if it's missing a newline.

File: ws/ws.go (hunk 1 above)
Comment (victorbalan) [inline ws/ws.go]: yes, I tried an that's the problem.

Comment (flowlo) [inline ws/ws.go]: Well, that implies that the infinite loop breaks, which means there is an error returned on `ws.ReadMessage()`. Maybe that will tell us the root cause.

**Comment by flowlo** [comment]
Using strings for `compare` can now be reverted.
